File: Java/c) Fixed Heap Dynamic.java
import java.util.Scanner;

public class FixedHeapDynamic {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Fixed heap dynamic array - allocated on heap but size fixed after creation
        int[] arr = new int[5];
        
        System.out.println("Enter 5 elements:");
        for(int i = 0; i < 5; i++) {
            arr[i] = scanner.nextInt();
        }
        
        System.out.println("Fixed heap dynamic array in Java:");
        for(int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();
        
        scanner.close();
    }
}

File: C++/c) Fixed Heap Dynamic.cpp
#include <iostream>
using namespace std;

int main() {
    // Fixed heap dynamic array - allocated on heap but size fixed after creation
    int* arr = new int[5];

    cout << "Enter 5 elements:" << endl;
    for(int i = 0; i < 5; i++) {
        cin >> arr[i];
    }

    cout << "Fixed heap dynamic array in C++:" << endl;
    for(int i = 0; i < 5; i++) {
        cout << arr[i] << " ";
    }
    cout << endl;

    delete[] arr; // Manual memory management required
    return 0;
}
